### PR TITLE
romio/daos: Fix some compiler warnings

### PR DESCRIPTION
**Pull request by mchaarawi**
- update configure to look in lib64 for cart, since DAOS 0.9 installs cart separately but 1.0 installs it with DAOS libs
- remove some dead code.
- fix some build warnings.

## Author Checklist
* [ ] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [ ] Remove xfail from the test suite when fixing a test
* [ ] Commits are self-contained and do not do two things at once
* [ ] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [ ] Passes whitespace checkers
* [ ] Passes warning tests
* [ ] Passes all tests
* [ ] Add comments such that someone without knowledge of the code could understand
* [ ] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

File: src/mpi/romio/configure.ac [hunk 1]
```diff
@@ -784,7 +784,7 @@ AC_ARG_WITH([cart],
 
 AS_IF([test "x$with_cart" != xno], [
     CART="yes"
-    LDFLAGS="$LDFLAGS -L$with_cart/lib"
+    LDFLAGS="$LDFLAGS -L$with_cart/lib -L$with_cart/lib64"
```
Comment (pavanbalaji) [inline src/mpi/romio/configure.ac:787]: This entire thing should be replaced with the `PAC_SET_HEADER_LIB_PATH` macros, for consistency.

File: src/mpi/romio/adio/ad_daos/ad_daos_common.c [hunk 2]
```diff
@@ -1,22 +1,20 @@
...
- * GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
- * The Government's rights to use, modify, reproduce, release, perform, display,
- * or disclose this software are subject to the terms of the Apache License as
- * provided in Contract No. 8F-30005.
- * Any reproduction of computer software, computer software documentation, or
- * portions thereof marked with this legend must also reproduce the markings.
+ *  Portions of this code were written by Intel Corporation.
+ *  Copyright (C) 2018-2020 Intel Corporation.  Intel provides this material
+ *  to Argonne National Laboratory subject to Software Grant and Corporate
+ *  Contributor License Agreement dated February 8, 2012.
```
Comment (pavanbalaji) [inline src/mpi/romio/adio/ad_daos/ad_daos_common.c]: All of the contribution copyright material is being moved to the top-level copyright.  Specific CLA related information (such as the one above) should be maintained in the PR, not in the files.

File: src/mpi/romio/adio/ad_daos/ad_daos_common.c [hunk 3]
```diff
@@ -1,22 +1,20 @@
...
  */
 
 #include "ad_daos.h"
 #include <daos_errno.h>
 
 int ADIOI_DAOS_Initialized = MPI_KEYVAL_INVALID;
 
-int ADIOI_DAOS_End(MPI_Comm comm, int keyval, void *attribute_val, void *extra_state)
+static int ADIOI_DAOS_End(MPI_Comm comm, int keyval, void *attribute_val, void *extra_state)
```
Comment (pavanbalaji) [inline src/mpi/romio/adio/ad_daos/ad_daos_common.c:17]: `static` functions should not be namespaced.

**Comment by pavanbalaji** [comment]
@mchaarawi Your name isn't listed on the Intel CLA, so we cannot merge this PR till that is fixed.  Intel is working on it.

Tagging @wesbland and @ddurnov too.

File: src/mpi/romio/adio/ad_daos/ad_daos_common.c (hunk 2 above)
Comment (mchaarawi) [reply src/mpi/romio/adio/ad_daos/ad_daos_common.c]: hmm those are already in your master.
I guess you are not looking at the squashed changes. maybe i can squash this commit if that helps.

Comment (pavanbalaji) [reply src/mpi/romio/adio/ad_daos/ad_daos_common.c]: No, the move is not in master yet.  It's in a PR waiting for confirmation from our legal team.

File: src/mpi/romio/configure.ac (hunk 1 above)
Comment (mchaarawi) [reply src/mpi/romio/configure.ac:787]: > @mchaarawi Your name isn't listed on the Intel CLA, so we cannot merge this PR till that is fixed. Intel is working on it.
> 
> Tagging @wesbland and @ddurnov too.

ok let me rework these changes and repush a squashed PR when I'm added to the CLA. I know the process was started.

**Pull request closed without merging**